projects/packages/config/README.md

```diff
@@ -107,6 +107,25 @@ add its slug to the config class property:
     );
 ```
 
+## The ensure_options call
+
+Each consumer will initialize your package from its own instance of the `Config` class, and each consumer can call `$config->ensure( 'your-feature', $options )` passing different options.
+
+Your package will need to handle these options and decide what to do with them when different consumers pass diferent options.
+
+You do that by creating a `ensure_options_{$package_slug}` method to the `Config` class. For example:
+
+```
+
+public function ensure_options_configurable_package() {
+    $options = $this->get_feature_options( 'configurable_package' );
```

Comment:
> The downside of that is that we're passing the burden on to the package, but we are already doing it for package initialization code.

I think that's the best thing to do. Let the packages themselves decide what to do.

> What if we did it like a filter with a closure? Sort of like

I don't think Config should merge the settings. We should let packages handle them. A couple of examples why I think that:

If you look at what Connection does for example, it does not merge the options. It adds a new `Plugin` for each consumer that initializes it. It uses this rather as a hook to create a list of all consumers that are using the Connection.

Also, how we are going to merge the options? For example, if a consumer says that options A is `true` and another consumer says that option B is `false`.

Only the package will know what value should "win", depending on its business logic.

For the VideoPress package, for example, if any consumer sets `admin_ui` as `true`, I want it to override any other places where it might be set to `false`. This is a case by case decision that only the package knows how to handle.

> This is a great start, but what do you think about passing the options directly into the called method?

We could do that and avoid one function call... but it's good to store the options in the instance as we do now so it's also available in the `enable_$feature` method

Reply:
if you look at how Sync merge the options for example, it is not straight forward... many specific rules there...

Reply:
OK, so I think I misunderstood your intention back there. Now I see you would like run the options initialization at the time of feature declaration, not at the time the feature gets "ensured" at the `plugins_loaded` time.

I agree with you that Config shouldn't merge options, and that more complicated logic is sometimes involved in deciding what option "wins". Config was initially considered an "opt-in" mechanism, not really an "opt-in and configure", which it is now becoming. Let's try to go this route and see where it comes.

This does let us to move the custom Config code into packages, so maybe we'll look at that later, so that the logic feels more at home in the package maintainer's hands.

All that to say - cool, thanks for doing this, let's try to go this route.